odoo/tests/common.py

```diff
@@ -470,3 +470,40 @@ def can_import(module):
         return False
     else:
         return True
+
+
+def set_tags(*tags):
+    """
+    A decorator to tag TestCase objects
+    Tags are stored in a set that can be accessed from a 'test_tags' attribute
+    Tags must not contains '+' or '-' signs
+    """
+    def tags_decorator(obj):
+        if not hasattr(obj, 'test_tags'):
+            obj.test_tags = set()
+        obj.test_tags.update(set(tags))
+        return obj
+    return tags_decorator
+
+
+def check_tags(obj, tags):
+    """
+    Check if object 'obj' has test_tags 'tags' defined
+    Obj is considered to have the tag 'standard' by default.
+    tags selection pico-language:
+    [+]tag_name: means that if the test has this tag, it have to be executed
+    -tag_name: means that if the test has this tag, it does not have to be executed
+    """
+    if not tags:
+        return True
+    if isinstance(tags, str):
+        tags = set([t.strip() for t in tags.split(',')])
+    no_test_tags = set([t[1:] for t in tags if t.startswith('-')])
+    to_test_tags = set([t.replace('+', '') for t in tags if not t.startswith('-')])
```

Comment:
No need for the inner list, and could use a set comprehension to avoid the set() call as well